runtime/Cpp/runtime/src/UnbufferedCharStream.cpp

```diff
@@ -101,7 +101,7 @@ size_t UnbufferedCharStream::LA(ssize_t i) {
     return EOF;
   }
 
-  if (_data[(size_t)index] == (uint32_t)EOF) {
+  if (_data[(size_t)index] == 0xFFFF) {
```

Comment:
Are you sure this is valid? anytime you touch anything dealing with characters we need to be very careful as Ben spent a lot of time on this. why are you converting to a 16-bit -1?